libraries/aws_ecr_repository_policy.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AwsEcrPolicy < AwsResourceBase
+  name 'aws_ecr_repository_policy'
+  desc 'Verifies policy settings for an AWS ECR Repository'
+
+  example "
+    describe aws_ecr_repository_policy(repository_name: 'my-repository') do
+      it { should exist }
+    end
+  "
+
+  def initialize(opts = {})
+    # Create a repository_name:<value> pair if the argument is a string object.
+    opts = { repository_name: opts } if opts.is_a?(String)
+
+    super(opts)
+    validate_parameters(required: %i(repository_name))
+    @repo_name = { "repository_name": opts[:repository_name] }
+    @display_name = opts[:repository_name]
+    catch_aws_errors do
+      resp = @aws.ecr_client.get_repository_policy(@repo_name)
+      @repo_policy = resp.policy_text
+    end
+  end
+
+  def exists?
+    !@repo_policy.nil? && !@repo_policy.empty?
+  end
+
+  def has_statement?(criteria = {})
+    return false unless @repo_policy
+    statement_match = []
+    criteria = criteria.each_with_object({}) { |(k, v), h| h[k.to_sym] = v }
+    criteria[:Principal] = criteria[:Principal].is_a?(Array) ? criteria[:Principal].sort : criteria[:Principal]
+    statements.any? do |s|
+      actions = s[:Action] || []
+      effect = s[:Effect]
+      sid = s[:Sid]
+      principal = s[:Principal].is_a?(Array) ? s[:Principal].sort : s[:Principal]
+      action_match = criteria[:Action].nil? ? true : actions.include?(criteria[:Action])
+      effect_match = criteria[:Effect].nil? ? true : effect.eql?(criteria[:Effect])
+      sid_match = criteria[:Sid].nil? ? true : sid.eql?(criteria[:Sid])
+      principal_match = criteria[:Principal].nil? ? true : principal.eql?(criteria[:Principal])
+      statement_match.push(action_match && effect_match && sid_match && principal_match)
+    end
+    statement_match.include?(true)
```

Comment:
No this is not what I was expecting! please refer [this](https://github.com/inspec/inspec-aws/compare/adding_aws_cloudformation_stack_policy#diff-890f2f311b5cd91e8a14f01b459d4093adffdca1597a4d482259a31b9bb690a0R38-R52)

in your updated approach you check if statements are true twice! one with any? and another time with include?